cogito/protocol.go

```diff
@@ -0,0 +1,188 @@
+// This file implements the Concourse resource protocol described at
+// https://concourse-ci.org/implementing-resource-types.html
+
+package cogito
+
+import (
+	"encoding/json"
+	"fmt"
+	"io"
+	"os"
+	"strings"
+)
+
+// CheckInput is the JSON object passed to the stdin of the "check" executable plus
+// build metadata (environment variables).
+//
+// See https://concourse-ci.org/implementing-resource-types.html#resource-check
+//
+type CheckInput struct {
+	Source Source `json:"source"`
+	// Concourse will omit field Version from the first request.
+	Version Version `json:"version"`
+	Env     Environment
+}
+
+func NewCheckInput(in io.Reader) (CheckInput, error) {
+	var ci CheckInput
+
+	dec := json.NewDecoder(in)
+	dec.DisallowUnknownFields()
+	if err := dec.Decode(&ci); err != nil {
+		return ci, fmt.Errorf("parsing JSON from stdin: %s", err)
+	}
+
+	if err := ci.Source.Init(); err != nil {
+		return ci, err
+	}
+
+	// We don't validate the presence of field Version because, only for the check step,
+	// Concourse will omit it from the _first_ request.
+
+	ci.Env.Fill()
+
+	return ci, nil
+}
+
+// Source is the "source:" block in a pipeline "resources:" block for the Cogito resource.
+type Source struct {
+	//
+	// Mandatory
+	//
+	Owner       string `json:"owner"`
+	Repo        string `json:"repo"`
+	AccessToken string `json:"access_token"` // SENSITIVE
+	//
+	// Optional
+	//
+	GChatWebHook  string `json:"gchat_webhook"` // SENSITIVE
+	LogLevel      string `json:"log_level"`
+	LogUrl        string `json:"log_url"` // DEPRECATED
+	ContextPrefix string `json:"context_prefix"`
+}
+
+// String renders Source, redacting the sensitive fields.
+func (src Source) String() string {
+	var bld strings.Builder
+
+	fmt.Fprintln(&bld, "owner:         ", src.Owner)
+	fmt.Fprintln(&bld, "repo:          ", src.Repo)
+	fmt.Fprintln(&bld, "access_token:  ", redact(src.AccessToken))
```

Comment:
If I understand the question correctly, to use the secret value you use it as any other field: `fetchGit(src.AccessToken)`.

Function redact() (just below here) takes a value, not a pointer, so it will return a redacted copy of the string.